webargs/async.py

```diff
@@ -91,38 +95,55 @@ def use_args(self, argmap, req=None, locations=None, as_kwargs=False, validate=N
         def decorator(func):
             req_ = request_obj
 
-            @functools.wraps(func)
-            def wrapper(*args, **kwargs):
-                req_obj = req_
-
-                # if as_kwargs is passed, must include all args
-                force_all = as_kwargs
-
-                if not req_obj:
-                    req_obj = self.get_request_from_view_args(func, args, kwargs)
-                # NOTE: At this point, argmap may be a Schema, callable, or dict
-                parsed_args = yield from self.parse(argmap,
-                                                    req=req_obj, locations=locations,
-                                                    validate=validate, force_all=force_all)
-                if as_kwargs:
-                    kwargs.update(parsed_args)
-                    return func(*args, **kwargs)
-                else:
-                    # Add parsed_args after other positional arguments
-                    new_args = args + (parsed_args, )
-                    return func(*new_args, **kwargs)
+            if iscoroutinefunction(func):
+                @functools.wraps(func)
+                async def wrapper(*args, **kwargs):
+                    req_obj = req_
+
+                    # if as_kwargs is passed, must include all args
+                    force_all = as_kwargs
+
+                    if not req_obj:
+                        req_obj = self.get_request_from_view_args(func, args, kwargs)
+                    # NOTE: At this point, argmap may be a Schema, callable, or dict
+                    parsed_args = await self.parse(argmap,
```

Comment:
In order for builds to pass on python < 3.5, I think you'll need to add `# flake8: noqa` to this line.